Tests/NetworkingTests/SampleResponseProviderTests.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  SampleResponseProviderTests.swift
+//  
+//
+//  Created by Matej Molnár on 05.01.2023.
+//
+
+@testable import Networking
+import XCTest
+
+final class SampleResponseProviderTests: XCTestCase {
+    // swiftlint:disable:next force_unwrapping
+    private lazy var mockUrlRequest = URLRequest(url: URL(string: "https://reqres.in/api/users?page=2")!)
```

Comment:
besides this I would like also have integration test which will cover the flow  ( mock data -> store it via StorageProcessor -> load via this Response provider and test that final data are the same as at the beginning. 
Right now we have two different objects which stores/read data from file system and would like to avoid any discrepancy in future

Reply:
I agree, but for this test flow we would need a new `ResponseProvider` because the `SampleResponseProvider` only loads data from Assets but `EndpointRequestStorageProcessor` stores data into file system.